query-translator/src/types.rs

```diff
@@ -88,28 +115,47 @@ impl QueryFragment for ColumnOrExpression {
             &Column(QualifiedAlias(ref table, ref column)) => {
                 out.push_identifier(table.as_str())?;
                 out.push_sql(".");
-                out.push_identifier(column.as_str())
+                push_column(out, column);
+                Ok(())
             },
             &Integer(i) => {
                 out.push_sql(i.to_string().as_str());
                 Ok(())
-            }
+            },
+            &Value(ref v) => {
+                out.push_typed_value(v)
+            },
         }
     }
 }
 
 impl QueryFragment for Projection {
     fn push_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
-        self.0.push_sql(out)?;
-        out.push_sql(" AS ");
-        out.push_identifier(self.1.as_str())
+        use self::Projection::*;
+        match self {
+            &One => out.push_sql("1"),
+            &Star => out.push_sql("*"),
+            &Columns(ref cols) => {
+                let &ProjectedColumn(ref col, ref alias) = &cols[0];
```

Comment:
There's [`intersperse`](https://docs.rs/itertools/0.5.9/src/itertools/intersperse.rs.html#13-19) for these situations...